packages/@o3r/create/src/index.ts

```diff
@@ -191,18 +203,38 @@ const prepareWorkspace = (relativeDirectory = '.', projectPackageManager = 'npm'
   writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2));
 
   if (projectPackageManager === 'yarn') {
-    exitProcessIfErrorInSpawnSync(2, spawnSync(runner, ['set', 'version', argv['yarn-version'] || 'stable'], {
-      stdio: 'inherit',
-      shell: true,
-      cwd
-    }));
+    exitProcessIfErrorInSpawnSync(YARN_SET_VERSION_ERROR_CODE, spawnSync(
+      runner,
+      ['set', 'version', argv['yarn-version'] || 'stable'],
```

Comment:
NB as not part of your code but it could be nice to add some checks over the yarn-version argument